torch/lib/c10d/ProcessGroupGloo.cpp

```diff
@@ -223,22 +223,23 @@ ProcessGroupGloo::ProcessGroupGloo(
   }
 
   threads_.resize(options.threads);
+  workInProgress_.resize(options.threads);
```

Comment:
Can you add a comment above saying why workInProgress needs to be the same as the size of number of threads, just to clarify to the code readers.